tests/dataset_test.py

```diff
@@ -0,0 +1,141 @@
+        self.assertEqual(ds.path, '/ipfs/Qm.path')
+        self.assertFalse(ds._is_populated)
+
+    def test_init_from_get(self):
+        loader.set_instance(mock_loader.DatasetMockLoader())
+        ds = dataset.Dataset(GET_OBJ)
+        self.assertEqual(ds.username, 'me')
+        self.assertEqual(ds.name, 'first_dataset')
+        self.assertEqual(ds.path, '/ipfs/Qm.path')
+
+        self.assertEqual(ds.body_path_value, "/ipfs/Qm.bodyPath")
+        self.assertEqual(ds.previous_path_value, "/ipfs/Qm.previousPath")
+
+        # Checking a subset of the possible fields for brevity
+
+        # TODO - should I test every field? I realized it would get pretty verbose but I can.
+        # I figured I'd test set_fields and move on.
+        # However if you want a super verbose and thorough comparison I can do that.
+        self.assertIsInstance(ds.commit_component, dataset.Commit)
+        self.assertEqual(ds.commit_component.author, {"id": "Qm.commit.author.id"})
+        self.assertEqual(ds.commit_component.message, "commit.message")
+        self.assertEqual(ds.commit_component.timestamp, "2020-01-02T03:04:05.124963898Z")
+
+        self.assertIsInstance(ds.meta_component, dataset.Meta)
+        self.assertEqual(ds.meta_component.description, 'meta.description')
+        self.assertEqual(ds.meta_component.keywords, ['meta.keyword1', 'meta.keyword2'])
+
+        self.assertIsInstance(ds.readme_component, dataset.Readme)
+        self.assertEqual(ds.readme_component.script.strip(), "TEST README")
+
+        self.assertIsInstance(ds.structure_component, dataset.Structure)
+        self.assertEqual(ds.structure_component.entries, 3)
+        self.assertEqual(ds.structure_component.format, 'csv')
+        self.assertEqual(ds.structure_component.schema, {
+            "items": {
+                "items": [
+                    {"title": "field1", "type": "string"},
+                    {"title": "field2", "type": "integer"},
+                ],
+                "type": "array"
+            },
+            "type": "array"
+        })
+
+        self.assertIsNone(ds.body_component)
+
+        self.assertTrue(ds._is_populated)
+
+    def test_populate(self):
+        loader.set_instance(mock_loader.DatasetMockLoader(get_responses={
+            'me/first_dataset': GET_OBJ
+        }))
+        ds = dataset.Dataset(LIST_OBJ)
+        self.assertFalse(ds._is_populated)
+        self.assertFalse(hasattr(ds, 'meta_component'))
+        ds.meta
+        self.assertTrue(ds._is_populated)
+        self.assertTrue(hasattr(ds, 'meta_component'))
+        self.assertEqual(ds.meta.description, 'meta.description')
+
+    def test_dont_populate_twice(self):
+        loader.set_instance(mock_loader.DatasetMockLoader())
+        ds = dataset.Dataset(GET_OBJ)
+        self.assertTrue(ds._is_populated)
+        self.assertTrue(hasattr(ds, 'meta_component'))
+        ds.meta # The mock loader should complain if it calls "get"
```

Comment:
To make it read better, I would change this comment to something that states what is actively being tested, rather than what isn't happening. Something like:

```
Ensure that `get` is not called when accessing `meta`, which would cause the [EventualName]Loader to throw.
```

Also, place it above the line in question, rather than after